server/service/instance.go

```diff
@@ -661,22 +655,54 @@ func (s *InstanceService) Find(ctx context.Context, in *pb.FindInstancesRequest)
 		if provider != nil {
 			err = serviceUtil.AddServiceVersionRule(ctx, domainProject, service, provider)
 		} else {
-			mes := fmt.Errorf("%s failed, provider does not exist", findFlag())
+			mes := fmt.Errorf("%s failed, provider does not exist", findFlag)
 			log.Errorf(mes, "AddServiceVersionRule failed")
 			return &pb.FindInstancesResponse{
 				Response: proto.CreateResponse(scerr.ErrServiceNotExists, mes.Error()),
 			}, nil
 		}
 		if err != nil {
-			log.Errorf(err, "AddServiceVersionRule failed, %s failed", findFlag())
+			log.Errorf(err, "AddServiceVersionRule failed, %s failed", findFlag)
 			return &pb.FindInstancesResponse{
 				Response: proto.CreateResponse(scerr.ErrInternal, err.Error()),
 			}, err
 		}
 	}
 
+	return s.genFindResult(ctx, rev, item)
+}
+
+func (s *InstanceService) findSharedServiceInstance(ctx context.Context, in *pb.FindInstancesRequest, provider *pb.MicroServiceKey) (*pb.FindInstancesResponse, error) {
+	var err error
+	service := &pb.MicroService{Environment: in.Environment}
+	// it means the shared micro-services must be the same env with SC.
+	provider.Environment = apt.Service.Environment
+	findFlag := fmt.Sprintf("find shared provider[%s/%s/%s/%s]", provider.Environment, provider.AppId, provider.ServiceName, provider.Version)
+
+	// cache
+	var item *cache.VersionRuleCacheItem
+	rev, _ := ctx.Value(util.CtxRequestRevision).(string)
```

Comment:
check if this value is string，never ignore exception

Reply:
Done